src/sdk/sdk.test.ts

```diff
@@ -253,21 +259,21 @@ describe("sanitizeResourceName", () => {
   it("should replace sequences of non-alphanumeric characters with a single -", () => {
     const resourceName = "test-*^%- -!=!resource";
     const sanitizedResourceName = sanitizeResourceName(resourceName);
-    expect(sanitizedResourceName).toEqual("test-resource");
+    expect(sanitizedResourceName).toEqual("test------resource");
   });
 
-  it("should truncate name to 250 characters", () => {
+  it("should truncate name to 253 characters", () => {
     const resourceName =
-      "test-resourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresource";
+      "test-resourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourcemoreletters";
     const sanitizedResourceName = sanitizeResourceName(resourceName);
     expect(sanitizedResourceName).toEqual(
-      "test-resourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresourceresou",
+      "test-resourceresourceresourceresourceresourceresourceresourcere",
     );
   });
 
   it("should remove leading and trailing non-letter characters", () => {
     const resourceName = " 1=-test-resource *2 ";
     const sanitizedResourceName = sanitizeResourceName(resourceName);
-    expect(sanitizedResourceName).toEqual("test-resource");
+    expect(sanitizedResourceName).toEqual("1--test-resource-2");
```

Comment:
```suggestion
    expect(sanitizedResourceName).toEqual("1-test-resource-2");
```

Based off the test case `"should replace sequences of non-alphanumeric characters with a single -"` does that mean we should not have a `--` in the new resource name?

Reply:
you can, i did not realize how flexible K8s was with naming.